### PR TITLE
Budget App

### PR DESCRIPTION
**Pull request by SelmaNdapanda**
#### Design  
- You should follow these [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), including:  :heavy_check_mark: 
  - Colors.
  - Typography: font face, size and weight.
  - Layout: composition and space between elements.

#### Interactions  
- Splash screen  :heavy_check_mark: 
  - A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

- Sign up and log in pages  :heavy_check_mark: 
  - The user should be able to register in the app with full name, email and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- Home page (categories page)  :heavy_check_mark: 
  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page  :heavy_check_mark: 
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page  :heavy_check_mark: 
  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page  :heavy_check_mark: 
  - The user fills out a form to create a new transaction with:
    - name (mandatory)
    - amount (mandatory)
    - categories (mandatory at least one)
  - The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

#### Testing requirements
- Create unit and integration tests for all the most important components of your RoR application.  :heavy_check_mark: 

#### Technical requirements  

- You should use Postgres as your database.  :heavy_check_mark: 
- You should use devise for authentication.  :heavy_check_mark: 
- You should validate all user input to make sure that anyone with bad intentions cannot compromise your app.  :heavy_check_mark: 
- You can use a view template engine of your choice (.erb, .slim, .haml).  :heavy_check_mark: 
- The project should be deployed and accessible online.  :heavy_check_mark: 
- Your database schema should reflect the following structure:  :heavy_check_mark: 

<p align="center">
  ERD diagram
</p>

### CODE REVIEW

**Review by fobadara — COMMENTED** [review]
Hi @SelmaNdapanda, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, with a PR that wants to merge from development into mai instead of a feature branch into main Kindly check tis [link](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md) out

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by SelmaNdapanda** [comment]
Hi @fobadara, I have followed the link you sent me, but it refers to feature branches that need review, not to the capstone project. 
The development branch is the one with all my changes because this capstone project only requires one review for all the changes. 
The development branch was previously my default branch, hence, I merged all my feature branches into the development branch. I have followed Gitflow.

**Comment by SelmaNdapanda** [comment]
Or may you perhaps suggest a simpler way of doing this?

File: app/models/group.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+class Group < ApplicationRecord
+  belongs_to :user
+  has_many :transacs
+
+  validates :name, presence: true, uniqueness: true
+  validates :icon, presence: true
+
```
Comment (MeqdamAlqudah) [inline app/models/group.rb]: - [ ] Please as in the  diagram the group and transacs should have many to many relationship currently they have on to many relationship.
image

**Review by MeqdamAlqudah — CHANGES_REQUESTED** [review]
Hi @SelmaNdapanda 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
image


## Highlights
:+1: design 
:+1: figure most of the things from the first time

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
-[Optional] make the design better , I don't think that the color of the signout  button matches the design
image


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/group.rb (hunk 1 above)
Comment (SelmaNdapanda) [reply app/models/group.rb]: Thank you for the good review, I will implement it now

**Comment by fobadara** [comment]
> Or may you perhaps suggest a simpler way of doing this?

Sorry I didn't see this early enough. The link is on the instruction for setting up Gitflow in this project. When you are submitting a solo project you are meant to create a PR from a feature branch into the dev branch

File: config/routes.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+Rails.application.routes.draw do
+  devise_for :users
+  resources :splash, only: [:index]
+
+  resources :groups, only: [:index, :new, :create] do
+    resources :transacs, only: [:index]
```
Comment (BenMukebo) [inline config/routes.rb:6]: - [OPTIONAL] There should have more than one route (show, create, and new ) in the config/routes.rb file for transacs. Kindly let's add this. ✍🏻

File: spec/models/transac_spec.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
+require 'rails_helper'
+
```
Comment (BenMukebo) [inline spec/models/transac_spec.rb:2]: - [OPTIONAL] Excellent adding unit tests for all the most important components of the RoR application 👍🏻 Let's now add more integration tests with capybara. 👍🏻

**Review by BenMukebo — APPROVED** [review]
# Status: Approved  :champagne:  ✔️:confetti_ball: 

## Hi @SelmaNdapanda, 

### **Congratulations! 🎉 Project Approved ✅**

Good job so far. Well Done.👍🏻 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 


### To Highlight 🎉
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️
- Good presentation video. ✔️ 
- Your app has been well deployed heavy_check_mark
- Your Design looks awesome, keep rocking :rocket: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers :clinking_glasses:  and Happy coding!!! :dancers:💻  

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by SelmaNdapanda** [comment]
Thank you so much for approving my project. I will continue to implement the optional changes

**Pull request merged**